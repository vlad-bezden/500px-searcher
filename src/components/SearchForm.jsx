import React from 'react'

class SearchForm extends React.Component {
  constructor() {
    super()
    
    this.state = {
      query: ''
    }
  }
  
  static propTypes = {
    onSearch: React.PropTypes.func.isRequired
  }

	handleSubmit = (e) => {
    e.preventDefault()
    this.props.onSearch(this.state.query)
  }
  
  render() {
    return (
      <form className='form-inline' onSubmit={this.handleSubmit}>
        <div className='form-group'>
          <input
            type='text'
            className='form-control'
            id='search'
            placeholder='Enter your search'
            style={{ marginRight: '5px'}}
            onChange={(e) => this.setState({query: e.target.value})}
          />
        </div>
        
        <button
          type='submit'
          className='btn btn-default'>
          Search
        </button>
      </form>
    )
  }
}

export default SearchForm