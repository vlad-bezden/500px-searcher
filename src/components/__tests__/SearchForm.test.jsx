import React from 'react'
import renderer from 'react-test-renderer'

import SearchForm from '../SearchForm'

test('SearchForm renders correctly', () => {
    const component = renderer.create(
        <SearchForm onSearch={() => {}} />
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})