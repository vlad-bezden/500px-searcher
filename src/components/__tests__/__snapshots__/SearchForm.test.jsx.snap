// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<form
  className="form-inline"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="search"
      onChange={[Function]}
      placeholder="Enter your search"
      style={
        Object {
          "marginRight": "5px",
        }
      }
      type="text"
    />
  </div>
  <button
    className="btn btn-default"
    type="submit"
  >
    Search
  </button>
</form>
`;
